sql/auth/role_tables.cc

```diff
@@ -331,5 +334,26 @@ bool populate_roles_caches(THD *thd, TABLE_LIST *tablelst) {
     opt_mandatory_roles_cache = false;
   }
 
+    // Re-grant external roles provided by authentication plugins
+    for(const auto& p: g_external_roles) {
+      const char* host = p.first.second.c_str();
+      const char* username = p.first.first.c_str();
+      ACL_USER* acl_user = find_acl_user(host, username, true);
+      if (acl_user == nullptr) {
+        continue;
+      }
+
+      for(const auto& r: p.second) {
+        const char* host = r.second.c_str();
```

Comment:
are we re-declaring host here?

Reply:
actually yes, cannot compile:
```
/work/ps/src/8.0/sql/auth/role_tables.cc:347:21: error: declaration of ‘host’ shadows a previous local [-Werror=shadow=compatible-local]
         const char* host = r.second.c_str();
                     ^~~~
/work/ps/src/8.0/sql/auth/role_tables.cc:339:19: note: shadowed declaration is here
       const char* host = p.first.second.c_str();
```

Reply:
I compiled this without maintainer mode - it shadows the previous declaration, but it is a different variable. I'll fix that, but it wasn't critical for a prototype.